fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/impl/AbstractPersister.java

```diff
@@ -84,13 +72,45 @@ protected static ResourceHeaders readHeaders(final OCFLObjectSession objSession,
 
     @Override
     public boolean handle(final ResourceOperation operation) {
-        //ensure that at least one of them match.
-        for (final var c : this.resourceOperations) {
-            if (c.isInstance(operation)) {
-                //return true if the operation types match.
-                return this.resourceOperationType.contains(operation.getType());
-            }
+            return resourceOperation.isInstance(operation) && resourceOperationType.equals(operation.getType());
+    }
+
+    protected FedoraOCFLMapping getMapping(final String resourceId) throws PersistentStorageException {
+        try {
+            return this.index.getMapping(resourceId);
+        } catch (FedoraOCFLMappingNotFoundException e){
+            throw new PersistentStorageException(e.getMessage());
         }
-        return false;
     }
+
+    protected FedoraOCFLMapping findMapping(final ResourceOperation operation,
```

Comment:
Needs javadoc

Reply:
I don't think we have a policy. I was just trying to be able to understand this code if I have to work with it. As long as you and @dbernstein understand how this all works then I'm fine.

Reply:
only having two people able to understand the code doesn't seem like a good goal to me, so i'm in favor of adding documentation